<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LoginDialog</class>
 <widget class="QDialog" name="LoginDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Вхід до системи</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QDialog#LoginDialog {
    background-color: #f8f9fa; /* Світло-сірий фон */
}

QLabel {
    color: #212529; /* Темний текст */
    font-size: 10pt;
}

QLabel#titleLabel {
    font-size: 16pt;
    font-weight: bold;
    color: #007bff; /* Синій заголовок */
    margin-bottom: 15px;
	margin-top: 10px;
}

QLabel#errorLabel {
    color: #dc3545; /* Червоний для помилок */
    font-size: 9pt;
    font-weight: bold;
	min-height: 20px; /* Резервуємо місце */
}

QLineEdit {
    padding: 8px;
    border: 1px solid #ced4da; /* Світло-сіра рамка */
    border-radius: 4px;
    background-color: #ffffff; /* Білий фон */
    font-size: 10pt;
}

QLineEdit:focus {
    border-color: #80bdff; /* Синя рамка при фокусі */
    /* Додатково можна додати тінь, якщо бажаєте */
    /* box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.25); */
}

QPushButton {
    padding: 8px 15px;
    border: 1px solid #007bff; /* Синя рамка */
    border-radius: 4px;
    background-color: #007bff; /* Синій фон */
    color: white; /* Білий текст */
    font-size: 10pt;
	font-weight: bold;
    min-width: 90px; /* Мінімальна ширина */
}

QPushButton:hover {
    background-color: #0056b3; /* Темніший синій при наведенні */
    border-color: #0056b3;
}

QPushButton:pressed {
    background-color: #004085; /* Ще темніший синій при натисканні */
    border-color: #00376e;
}

QPushButton:disabled {
    background-color: #6c757d; /* Сірий, коли вимкнено */
    border-color: #6c757d;
    color: #e9ecef;
}

QPushButton#cancelButton {
    background-color: #6c757d; /* Сіра кнопка скасування */
    border-color: #6c757d;
}

QPushButton#cancelButton:hover {
    background-color: #5a6268;
    border-color: #545b62;
}

QPushButton#cancelButton:pressed {
    background-color: #545b62;
    border-color: #4e555b;
}

/* Кнопки перемикання режимів (стиль як посилання) */
QPushButton#switchToRegisterButton, QPushButton#switchToLoginButton {
    background-color: transparent;
    border: none;
    color: #007bff; /* Синій текст */
    text-decoration: underline;
    padding: 2px;
    min-width: 0; /* Скидання мін. ширини */
	font-size: 9pt;
	font-weight: normal; /* Звичайний шрифт */
}

QPushButton#switchToRegisterButton:hover, QPushButton#switchToLoginButton:hover {
    color: #0056b3;
}

QPushButton#switchToRegisterButton:pressed, QPushButton#switchToLoginButton:pressed {
    color: #004085;
}

/* Забираємо стиль flat, бо він може конфліктувати */
QPushButton[flat=&quot;true&quot;] {
   /* Стилі вже визначені вище */
}

QFormLayout {
    vertical-spacing: 12px; /* Відстань між рядками форми */
	horizontal-spacing: 10px; /* Відстань між міткою та полем */
	margin-top: 15px;
}
</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="titleLabel">
     <property name="font">
      <font>
       <pointsize>14</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Вхід до Книгарні</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QFormLayout" name="formLayout">
     <property name="labelAlignment">
      <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
     </property>
     <item row="0" column="0">
      <widget class="QLabel" name="emailLabel">
       <property name="text">
        <string>Email:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="emailLineEdit">
       <property name="placeholderText">
        <string>Введіть ваш email</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="passwordLabel">
       <property name="text">
        <string>Пароль:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="passwordLineEdit">
       <property name="echoMode">
        <enum>QLineEdit::EchoMode::Password</enum>
       </property>
       <property name="placeholderText">
        <string>Введіть ваш пароль</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="firstNameLabel">
       <property name="text">
        <string>Ім'я:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="firstNameLineEdit">
       <property name="placeholderText">
        <string>Введіть ваше ім'я</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="lastNameLabel">
       <property name="text">
        <string>Прізвище:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="lastNameLineEdit">
       <property name="placeholderText">
        <string>Введіть ваше прізвище</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="confirmPasswordLabel">
       <property name="text">
        <string>Підтвердіть пароль:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLineEdit" name="confirmPasswordLineEdit">
       <property name="echoMode">
        <enum>QLineEdit::EchoMode::Password</enum>
       </property>
       <property name="placeholderText">
        <string>Введіть пароль ще раз</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="errorLabel">
     <property name="styleSheet">
      <string notr="true">color: #dc3545; /* Червоний для помилок */
font-size: 9pt;</string>
     </property>
     <property name="text">
      <string>Error message placeholder</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="switchButtonLayout">
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="switchToRegisterButton">
       <property name="text">
        <string>Немає акаунту? Зареєструватися</string>
       </property>
       <!-- Атрибут flat більше не потрібен, керуємо стилем через QSS -->
       <!-- <property name="flat">
        <bool>true</bool>
       </property> -->
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="switchToLoginButton">
       <property name="text">
        <string>Вже є акаунт? Увійти</string>
       </property>
       <!-- Атрибут flat більше не потрібен -->
       <!-- <property name="flat">
        <bool>true</bool>
       </property> -->
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Policy::Expanding</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="cancelButton">
       <property name="text">
        <string>Скасувати</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="okButton">
       <property name="text">
        <string>Увійти</string>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>cancelButton</sender>
   <signal>clicked()</signal>
   <receiver>LoginDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>267</x>
     <y>316</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>174</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
